Modules/_asynciomodule.c

```diff
@@ -3345,9 +3371,16 @@ _asyncio__register_task_impl(PyObject *module, PyObject *task)
 /*[clinic end generated code: output=8672dadd69a7d4e2 input=21075aaea14dfbad]*/
 {
     asyncio_state *state = get_asyncio_state(module);
-    if (register_task(state, task) < 0) {
-        return NULL;
+    if (!Task_Check(state, task)) {
```

Comment:
would be useful to add a comment here explaining that "native asyncio tasks" (or 3rd party tasks that inherit from native asyncio tasks) are managed using a (more efficient) linked list, but other types of tasks fallback to the (less efficient) weakset

Reply:
I have added some comments and  explanation for state of linked list.